controllers/topology/reconcile_state.go

```diff
@@ -144,12 +147,21 @@ func (r *ClusterReconciler) reconcileMachineDeployments(ctx context.Context, s *
 	}
 
 	// Update MachineDeployments.
+	// update at most MachineDeploymentUpdateLimit machine deployments
+	updateCounter := 0
 	for _, mdTopologyName := range diff.toUpdate {
+		if updateCounter >= machineDeploymentUpdateLimit {
+			break
+		}
```

Comment:
We should resolve only one during compute, because doing this during reconcile will prevent to apply any other change to a machine deployment while another machine deployment is upgrading.
Also, I'm not entirely sure by the logic below, because it seems to me that we are doing a single update during each reconcile, but nothing prevents to start the second upgrade during the next reconcile no matter if the first reconcile completed or not